pkg/cmd/api/api.go

```diff
@@ -233,6 +270,7 @@ func NewCmdApi(f *cmdutil.Factory, runF func(*ApiOptions) error) *cobra.Command
 	cmd.Flags().StringArrayVarP(&opts.RequestHeaders, "header", "H", nil, "Add a HTTP request header in `key:value` format")
 	cmd.Flags().StringSliceVarP(&opts.Previews, "preview", "p", nil, "GitHub API preview `names` to request (without the \"-preview\" suffix)")
 	cmd.Flags().BoolVarP(&opts.ShowResponseHeaders, "include", "i", false, "Include HTTP response status line and headers in the output")
+	cmd.Flags().BoolVar(&opts.Slurp, "slurp", false, "Use with \"--paginate\" to return an array of all pages of either JSON arrays or objects")
```

Comment:
Sorry @heaths, one of the issues with long running PRs like this is remembering all the context. When I played around with this just now I anticipated that for REST endpoints `--paginate --slurp` would be the same as `--paginate` since the content of REST responses already get slurped into an array. However what happens is we end up doubly nesting the array:

```
[
  [
    {
      "url": "https://api.github.com/repos/cli/cli/releases/149628351",
...
    }
  ]
]
```


Maybe my assumption is bad and we always want to apply the same semantics to `--slurp` even if it produces a strange result - we already know that the whole pagination is kind of quirky. What do you think?

Reply:
That was how it originally worked, but then someone noted it didn't "solve" their problem of merging nested arrays - only top-level arrays. I still think that's up to business logic - there's no way for us to know which level of arrays to merge - but in the discussion above I agreed to just wrap everything in an array like `jq --slurp` so that we at least return valid JSON. That's all this PR has ever been about: returning valid JSON such that you can pipe it to something that requires it - not `jq`, but something like PowerShell's `ConvertFrom-Json`.

Reply:
I remember that discussion as it related to the [GQL requests](https://github.com/cli/cli/pull/8620#issuecomment-1947020381) and I agree that we cannot know that and the consumer needs to determine that themselves.

To be clear, here I'm describing the existing merging of JSON array REST responses already getting slurped by `--paginate` with no further work (as far as I can tell). Below in the screenshot you can see the difference between `./bin/gh api --paginate repos/cli/go-gh/release` and `./bin/gh api --paginate --slurp repos/cli/go-gh/release`. The `--slurp` results in a double nesting of arrays with the top level one only having one item.

It's just a bit of a weird situation that the REST pagination ends up creating a single array object that the slurp then wraps in another array. I'm not sure there is a good solution to this though, except perhaps only allowing `--slurp` on graphql paginated requests? In any case someone is liable to get confused, so maybe we just ship this and move on with our lives.

<img width="2453" alt="image" src="https://github.com/cli/cli/assets/1611510/a99f3c31-1be6-4523-a1e2-b38aef2992f5">

Reply:
Yes, that was the compromised we discussed in our meeting with @andyfeller as well: we'd mimic `jq --slurp` exactly, which also double-nests arrays. This way, the result is consistent: it's always an array. The array items are each page of the response regardless of whether it's an array or object. No guesswork as to how best to handle the output: it's always an array of each page. If you don't need that for a REST response of arrays, don't pass `--slurp`: you'll already get a merged array because of Nate's previous change based on my old PR.

I'd rather have consistency than conditional output formats myself. Easier to code without having to think of whether I'm making a REST call or GraphQL call.